tests/ext-popup.test.js

```diff
@@ -188,20 +188,11 @@ test('Error is shown in popup UI while wrong parameter is passed to handleEvent'
 });
 
 test('clicking on "view activity logs page" button should open the activitylog page', async () => {
-  document.body.innerHTML = popupBody;
-
-  const getMonitorStatusFn = jest.spyOn(ExtListen, 'getMonitorStatus');
   const openActivityLogPageFn = jest.spyOn(ExtListen, 'openActivityLogPage');
 
-  getMonitorStatusFn.mockResolvedValue(false);
-  openActivityLogPageFn.mockResolvedValue();
-
   const popup = new Popup();
   await popup.init();
+  popup.openActivityLogBtn.click();
 
-  const openActivityLogBtn = popup.openActivityLogBtn;
-
-  await popup.init();
-  openActivityLogBtn.click();
   expect(openActivityLogPageFn).toHaveBeenCalled();
 });
```

Comment:
This is currently passing mainly because the tests before it are making sure that the popup will be mocking the getMonitorStatus method, but if you try to change `test(...)` into `test.only(...)` (which will run only that test and skip the other ones) the test will be failing because it is going to execute the real getMonitorStatus method which is internally calling WebExtensions API (which are not available in jsdom test environment unless we have injected them).

I looked into what was triggering the failure and it was triggered from inside the jest teardown phase, because it tries to access properties of the window (window.location in particular) for a mocked jsdom window that was apparently gone.

Turned out that is likely because a previous jsdom version didn't implement `window.close` and then it got implemented in a more recent jsdom version used by the updated jest dependency in this PR, the `window.close` is called from inside the click handler (to autoclose the browserAction popup) that is expected to be triggered by this test (which is why this was the only test triggering it).

We can fix that by mocking window.close in this test case.

Some on the cleanup you have done in this version of the fixup patch is still worth it, but parts currently removed are meant to stay as mentioned at the start of this comment.

Would you mind to update the last patch to make this test to look like in the following snippet? 
```
test('clicking on "view activity logs page" button should open the activitylog page', async () => {
  document.body.innerHTML = popupBody;

  const getMonitorStatusFn = jest.spyOn(ExtListen, 'getMonitorStatus');
  const openActivityLogPageFn = jest.spyOn(ExtListen, 'openActivityLogPage');

  getMonitorStatusFn.mockResolvedValue(false);
  openActivityLogPageFn.mockResolvedValue();
  // Prevent window.close to trigger a failure due
  // to jest teardown not expecting the jsdom window
  // to be gone.
  window.close = jest.fn(() => {});

  const popup = new Popup();
  await popup.init();
  popup.openActivityLogBtn.click();
  expect(openActivityLogPageFn).toHaveBeenCalled();
  // Expect the browserAction popup to autoclose itself.
  expect(window.close).toHaveBeenCalled();
});
```

After the PR is updated and the CI job confirmed that the tests pass successfully (I tried that locally, but better to confirm that will stay the case also on the CI), you can proceed and merge the patch, but I'm also happy to look to the updated version and explicitly approve it on my side.

Reply:
Thank you!